### PR TITLE
React PWA example

### PR DESCRIPTION
**Pull request by satyambnsal**
Closes #
#80 
## Introduced changes
A basic react PWA example that can be installed as a desktop or mobile app from browsers.

## Checklist

-   [x] Linked relevant issue
-   [x] Updated relevant documentation
-   [x] Added relevant tests
-   [ ] Add a dedicated CI job for new examples
-   [x] Performed self-review of the code

### CODE REVIEW

**Comment by satyambnsal** [comment]
<div>
    <a href="https://www.loom.com/share/45f49e263aeb4463986a94531aa78607">
      <p>React PWA example walkthrough - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/45f49e263aeb4463986a94531aa78607">
      
    </a>
  </div>

File: examples/react/react-pwa-app/readme.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+## Instructions
+
+### How to generate icon assets
+
+1. To generate image icon assets, first replace `public/logo.svg` with your own logo image. You can also change `pwa-assets.config.ts` to use logo with different name or path.
+2. Then run `npm run generate-pwa-assets` to generate the assets
+
+## Known issues
+
+1. Bun runtime might fail while running generate assets script as their is an [open issue](https://github.com/vite-pwa/assets-generator/issues/38) with [@vite-pwa/assets-generator](https://github.com/vite-pwa/assets-generator)
```
Comment (ponderingdemocritus) [inline examples/react/react-pwa-app/readme.md:18]: lets move to pnpm here

File: examples/react/react-pwa-app/src/assets/react.svg
Comment (ponderingdemocritus) [inline examples/react/react-pwa-app/src/assets/react.svg:1]: lets include the dojo logo here

**Comment by ponderingdemocritus** [comment]
Looking great!

Can you add it to the build step in the top level package.json and fix the other small things, then lets ship!

File: examples/react/react-pwa-app/readme.md (hunk 1 above)
Comment (satyambnsal) [reply examples/react/react-pwa-app/readme.md:18]: generate script works with any package manager **npm**, **yarn**, or **pnpm**. it doesn't work with **bun** only due to this issue. thought its best to document if user is trying with **bun** and facing the issue.

File: package.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
         "packages/state"
     ],
     "devDependencies": {
+        "@vite-pwa/assets-generator": "^0.2.4",
```
Comment (ponderingdemocritus) [inline package.json:16]: we don't need this at top level

**Review by ponderingdemocritus — APPROVED** [review]

**Pull request merged**